### PR TITLE
Update osgi.contract version to 6.0.0 and name to JakartaExpressionLanguage

### LINKED ISSUE #25869: Expression Language 6.0: Update osgi.contract to 6.0
As part of the initial delivery, we kept the version at `5.0` to avoid issues with other EE11 features that have not been updated to EE11 content yet. This initial work was done here: https://github.com/OpenLiberty/open-liberty/issues/25703

In addition we should ensure we use the new osgi.contract name once it is updated in the Tomcat Expression Language: https://bz.apache.org/bugzilla/show_bug.cgi?id=66834

### PR DESCRIPTION
**Pull request by pnicolucci**
fixes #25869

### CODE REVIEW

**Comment by pnicolucci** [comment]
#libby #build

**Comment by volosied** [comment]
Some apps failed to start because of 
```
[ERROR   ] CWWKE0702E: Could not resolve module: io.openliberty.cdi.4.0.internal.web [137]
  Unresolved requirement: Require-Capability: osgi.contract; osgi.contract="JavaEL"; filter:="(&(osgi.contract=JavaEL)(version=5.0.0))"

[ERROR   ] CWWKE0702E: Could not resolve module: com.ibm.ws.cdi.web.jakarta [138]
  Unresolved requirement: Import-Package: com.ibm.ws.cdi.web.factories; version="[1.0.0,2.0.0)"
    -> Export-Package: com.ibm.ws.cdi.web.factories; bundle-symbolic-name="io.openliberty.cdi.4.0.internal.web"; bundle-version="1.0.86.202401221656"; version="1.0.0"; uses:="com.ibm.websphere.ras.annotation,com.ibm.ws.ras.instrument.annotation,jakarta.el,jakarta.servlet,org.jboss.weld.manager,org.jboss.weld.module.web.servlet"
       io.openliberty.cdi.4.0.internal.web [137]
         Unresolved requirement: Require-Capability: osgi.contract; osgi.contract="JavaEL"; filter:="(&(osgi.contract=JavaEL)(version=5.0.0))"

```
This bundle uses 5.0 to build it: 

https://github.com/OpenLiberty/open-liberty/blob/3da58b960e1adbecda1ebdb414a2ecdaf5e5c33f/dev/io.openliberty.cdi.4.0.internal.web/bnd.bnd#L30

This is a general problem for any bundles used in EE11, but were build with EL 5.0 or lower.

**Comment by pnicolucci** [comment]
Yea looks like we need to wait for CDI to start doing some more EE11 work so their feature isn't using 4.0 code from EE10 anymore.

**Comment by wraschke** [comment]
#libby

**Comment by pnicolucci** [comment]
#libby #build

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
Need the new CDI bundle for:

```
[4/18/24, 14:53:22:487 EDT] 0000002f LogService-140-com.ibm.ws.cdi.web.jakarta                    E CWWKE0702E: Could not resolve module: com.ibm.ws.cdi.web.jakarta [140]
  Unresolved requirement: Import-Package: com.ibm.ws.cdi.web.factories; version="[1.0.0,2.0.0)"
    -> Export-Package: com.ibm.ws.cdi.web.factories; bundle-symbolic-name="io.openliberty.cdi.4.0.internal.web"; bundle-version="1.0.89.202404181412"; version="1.0.0"; uses:="com.ibm.websphere.ras.annotation,com.ibm.ws.ras.instrument.annotation,jakarta.el,jakarta.servlet,org.jboss.weld.manager,org.jboss.weld.module.web.servlet"
       io.openliberty.cdi.4.0.internal.web [139]
         Unresolved requirement: Require-Capability: osgi.contract; osgi.contract="JavaEL"; filter:="(&(osgi.contract=JavaEL)(version=5.0.0))"
```

**Comment by pnicolucci** [comment]
The remaining failures are for app security and auth I don't believe any EE11 work has started for these yet so we'll have to wait a bit longer.

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
```
8/8/24, 13:30:31:324 EDT] 0000002f gService-186-io.openliberty.security.jakartasec.3.0.internal E CWWKE0702E: Could not resolve module: io.openliberty.security.jakartasec.3.0.internal [186]
  Unresolved requirement: Require-Capability: osgi.contract; osgi.contract="JavaEL"; filter:="(&(osgi.contract=JavaEL)(version=5.0.0))"

[8/8/24, 13:30:31:325 EDT] 0000002f vice-187-io.openliberty.security.jakartasec.3.0.internal.cdi E CWWKE0702E: Could not resolve module: io.openliberty.security.jakartasec.3.0.internal.cdi [187]
  Unresolved requirement: Import-Package: io.openliberty.security.jakartasec.credential; version="[1.0.0,2.0.0)"
    -> Export-Package: io.openliberty.security.jakartasec.credential; bundle-symbolic-name="io.openliberty.security.jakartasec.3.0.internal"; bundle-version="1.0.93.202408081255"; version="1.0.0"; uses:="com.ibm.websphere.ras.annotation,com.ibm.ws.ras.instrument.annotation,io.openliberty.security.oidcclientcore.client,io.openliberty.security.oidcclientcore.token,jakarta.security.enterprise.credential,jakarta.security.enterprise.identitystore.openid,jakarta.servlet.http"
       io.openliberty.security.jakartasec.3.0.internal [186]
         Unresolved requirement: Require-Capability: osgi.contract; osgi.contract="JavaEL"; filter:="(&(osgi.contract=JavaEL)(version=5.0.0))"
  Unresolved requirement: Import-Package: io.openliberty.security.jakartasec; version="[1.0.0,2.0.0)"
    -> Export-Package: io.openliberty.security.jakartasec; bundle-symbolic-name="io.openliberty.security.jakartasec.3.0.internal"; bundle-version="1.0.93.202408081255"; version="1.0.0"; uses:="com.ibm.websphere.ras,com.ibm.websphere.ras.annotation,com.ibm.ws.ras.instrument.annotation,com.ibm.ws.security.javaeesec,io.openliberty.security.oidcclientcore.client,jakarta.security.enterprise.authentication.mechanism.http,jakarta.security.enterprise.authentication.mechanism.http.openid"
```

**Comment by pnicolucci** [comment]
#build

File: dev/io.openliberty.cdi.4.1.internal.web/.classpath [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="con" path="aQute.bnd.classpath.container"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
```
Comment (tevans78) [inline dev/io.openliberty.cdi.4.1.internal.web/.classpath]: I know this is a draft PR but ... not obvious to me why this new bundle is needed as there isn't any new code here? If it is needed, then this classpath entry should be Java 17?

Comment (tevans78) [reply dev/io.openliberty.cdi.4.1.internal.web/.classpath]: Java 17 should be set in bnd and other eclipse settings too

File: dev/io.openliberty.cdi.4.1.internal.web/bnd.bnd [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# SPDX-License-Identifier: EPL-2.0
+#*******************************************************************************
+-include= ~../cnf/resources/bnd/bundle.props
+bVersion=1.0
+
+Bundle-Name: io.openliberty.cdi.4.1.internal.web
+Bundle-SymbolicName: io.openliberty.cdi.4.1.internal.web
+Bundle-Description: CDI 4.1 Web, version ${bVersion}
+
+Import-Package: org.jboss.weld.*;version="[5.1,7)", \
```
Comment (tevans78) [inline dev/io.openliberty.cdi.4.1.internal.web/bnd.bnd]:
```suggestion
Import-Package: org.jboss.weld.*;version="[6.0,7)", \
```

File: dev/io.openliberty.cdi.4.1.internal.web/.classpath (hunk 1 above)
Comment (pnicolucci) [reply dev/io.openliberty.cdi.4.1.internal.web/.classpath]: Thanks @tevans78 on my cleanup list for this PR. The new bundle is needed to have a build dependency specifically on Jakarta Expression Language 6.0. If we don't have this bundle and we update the `osgi.contract` in `dev/io.openliberty.jakarta.expressionLanguage.6.0/bnd.bnd` then the following will occur:

https://github.com/OpenLiberty/open-liberty/pull/27428#issuecomment-2066412990

In this scenario, we need a new bundle that builds the code from the EE10 bundle with a dependency on Jakarta Expression Language 6.0.

File: dev/io.openliberty.cdi.4.1.internal.web/bnd.bnd (hunk 2 above)
Comment (pnicolucci) [reply dev/io.openliberty.cdi.4.1.internal.web/bnd.bnd]: I'll clean up this and the imports changed here: https://github.com/OpenLiberty/open-liberty/commit/31564b6bcc4955328f7118e6c513508aef496ff1#diff-70b4099a0254624a0439b86f34316cafc36693f5d0f8ae8daf6ff2a7e6959374 for the io.openliberty.cdi.4.0.internal.web bundle.

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
Latest build was good with just one failure: RTC defect 295756

File: dev/io.openliberty.security.jakartasec.3.0.internal/bnd.bnd [hunk 3]
```diff
@@ -35,11 +32,6 @@ Export-Package: \
     io.openliberty.security.jakartasec.tokens
 
 Import-Package: \
-    jakarta.security.enterprise.authentication.mechanism.http;version="[3.0,5)",\
-    jakarta.security.enterprise.authentication.mechanism.http.openid;version="[3.0,5)",\
-    jakarta.security.enterprise.credential;version="[3.0,5)",\
-    jakarta.security.enterprise.identitystore;version="[3.0,5)",\
-    jakarta.security.enterprise.identitystore.openid;version="[3.0,5)",\
```
Comment (pnicolucci) [inline dev/io.openliberty.security.jakartasec.3.0.internal/bnd.bnd:42]: This was added here: https://github.com/OpenLiberty/open-liberty/commit/3579c51b8c70b41bc6fc9b5b65a6be1855dc4f54#diff-1e3911bdae7806d50daacb0f68fa5cc4e8f447680ef8af2f16c6ec95b4dff9d1R38 so removing it now as it is not necessary.

File: dev/io.openliberty.cdi.4.0.internal.web/bnd.bnd [hunk 4]
```diff
@@ -17,9 +14,6 @@ Bundle-Name: io.openliberty.cdi.4.0.internal.web
 Bundle-SymbolicName: io.openliberty.cdi.4.0.internal.web
 Bundle-Description: CDI 4.0 Web, version ${bVersion}
 
-Import-Package: org.jboss.weld.*;version="[5.1,7)", \
```
Comment (pnicolucci) [inline dev/io.openliberty.cdi.4.0.internal.web/bnd.bnd:20]: This was added here: https://github.com/OpenLiberty/open-liberty/commit/31564b6bcc4955328f7118e6c513508aef496ff1#diff-70b4099a0254624a0439b86f34316cafc36693f5d0f8ae8daf6ff2a7e6959374R20 and I'm removing it now as it is no longer necessary

File: dev/io.openliberty.cdi.4.1.internal.web/bnd.bnd [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+# 
+# SPDX-License-Identifier: EPL-2.0
+#*******************************************************************************
+-include= ~../cnf/resources/bnd/bundle.props
+bVersion=1.0
+
+Bundle-Name: io.openliberty.cdi.4.1.internal.web
+Bundle-SymbolicName: io.openliberty.cdi.4.1.internal.web
+Bundle-Description: CDI 4.1 Web, version ${bVersion}
+
```
Comment (jhanders34) [inline dev/io.openliberty.cdi.4.1.internal.web/bnd.bnd:21]:
```suggestion

javac.source: 17
javac.target: 17

Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=17))"
```

File: dev/io.openliberty.security.jakartasec.4.0.internal/bnd.bnd [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+instrument.classesExcludes: io/openliberty/security/jakartasec/internal/resources/resources/*.class
+
+-buildpath: \
+    io.openliberty.security.jakartasec.3.0.internal;version=latest,\
+    com.ibm.ws.kernel.boot;version=latest,\
+    com.ibm.ws.kernel.service;version=latest,\
+    com.ibm.ws.logging;version=latest,\
+    com.ibm.ws.logging.core;version=latest,\
+    com.ibm.ws.org.osgi.annotation.versioning;version=latest,\
+    io.openliberty.security.jakartasec.2.0.internal,\
```
Comment (jhanders34) [inline dev/io.openliberty.security.jakartasec.4.0.internal/bnd.bnd:46]:
```suggestion
```
Is that one needed?

**Review by jhanders34 — CHANGES_REQUESTED** [review]

File: dev/io.openliberty.cdi.4.1.internal.web/bnd.bnd (hunk 5 above)
Comment (pnicolucci) [reply dev/io.openliberty.cdi.4.1.internal.web/bnd.bnd:21]: Thanks, I missed this as it wasn't in the original bnd.bnd for the 4.0 bundle! I'll update the 4.1 bundle.

File: dev/io.openliberty.security.jakartasec.4.0.internal/bnd.bnd (hunk 6 above)
Comment (pnicolucci) [reply dev/io.openliberty.security.jakartasec.4.0.internal/bnd.bnd:46]: @jhanders34 it was in the original bnd.bnd: https://github.com/OpenLiberty/open-liberty/blob/integration/dev/io.openliberty.security.jakartasec.3.0.internal/bnd.bnd#L53

Removing it causes the import in the manifest to change from: `com.ibm.ws.security.javaeesec;version="[1.0,2)"` to `com.ibm.ws.security.javaeesec`

**Comment by pnicolucci** [comment]
@jhanders34 requested change is pushed up let me know if you have any other comments and I'll start a final build before merging.

**Review by jhanders34 — APPROVED** [review]

**Comment by pnicolucci** [comment]
#libby #build

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#libby

**Comment by jhanders34** [comment]
#build

**Pull request merged**